src/tabpfn_common_utils/telemetry/core/decorators.py

```diff
@@ -260,19 +260,59 @@ def _make_callinfo(
     shapes: dict[str, tuple[int, ...]] = {}
     for param_name in param_names:
         if param_name in bound.arguments:
-            shape = shape_of(bound.arguments[param_name])
+            # Round the dimensionality of the dataset
+            shape = _round_dims(shape_of(bound.arguments[param_name]))
             if shape is not None:
                 shapes[param_name] = shape
 
     return _ModelCallInfo(shapes=shapes, task=task, model_method=model_method)
 
 
 def _infer_task(stack: list[_StackFrame]) -> ModelTaskType | None:
-    """Infer the model task from the call stack."""
+    """Infer the model task from the call stack.
+
+    Args:
+        stack: The call stack.
+
+    Returns:
+        The model task.
+    """
     for frame in stack:
         m = frame.module_name or ""
         if m.startswith("tabpfn.classifier"):
             return "classification"
         if m.startswith("tabpfn.regressor"):
             return "regression"
     return None
+
+
+def _round_dims(shape: tuple[int, int]) -> tuple[int, int]:
+    """Round the dimensionality of a dataset.
+
+    The intent is to anonymize the dataset dimensionality to prevent
+    leakage of sensitive information.
```

Comment:
```suggestion
    This function obscures the exact number of rows and columns in a dataset
    by rounding them up to the nearest predefined thresholds. This helps
    prevent leakage of sensitive information that might be inferred from
    precise dataset dimensions.

    Behavior:
    - Rows are rounded up to the nearest value in ``row_limits``.
      If larger than all thresholds, the row count is floored to the nearest
      multiple of 50.
    - Columns are rounded up to the nearest value in ``col_limits``.
      If larger than all thresholds, the column count is floored to the nearest
      multiple of 50.
    - If the input ``shape`` is empty or invalid, returns ``(0, 0)``.
```

I did find the function too trivial to read :-)

Reply:
True 😊 Let's make a compromise and balance between, agree that we don't want the documentation to be longer than the actual function so I slightly added more context.